File: Unidad 2-Estructuras de Control/Ejercicio24.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

/*
 *Autor: Alejandro Torre Reyes
Entradas: un numero n natural
Salida: la suma de 1/2^1 + 2/2^2...+ n/2^n
Proceso: Calcular la suma de los terminos 1/2^1 + 2/2^2...+ n/2^n
 */

namespace Ej24
{
    class Program
    {
        static int power(int bas, int x) //Metodo de potencia
        {
            int j;
            int p = 1;
            for(j=1; j<=x; j++)
            {
                p = p * bas;
            }
            return p;
        }
        static void Main(string[] args) //Clase Main
        {
            int n;
            int i;
            double sum = 0.0;
            int pot;
            double rec;
            Console.WriteLine("Ingresar n");
            n = Convert.ToInt32(Console.ReadLine());
            
            for(i=1; i<=n; i++) //Ciclo for para realizar la suma despues de cada iteracion
            {
                pot = power(2,i);
                rec = (1.0 / pot);
                sum = (sum + (rec * i));
            }

            Console.WriteLine("La suma es de {0}", sum); //Imprimir el resultado
            Console.ReadKey();
        }
    }
}
/* QA
Autor QA:Eyder Antonio Concha Moreno
Entradas: 1, 2, 10
salidas: .5 correcto, 1 correcto, 1.98828125 correcto
Proceso: Funciona correctamente
*/

File: Unidad 2-Estructuras de Control/ejercicio24.java
import java.util.Scanner;

/**
S.B.T.M
 */

class Ejercicio_30 {
  public static void main(String Args[]){

  /* suma de terminos de 1/2 + 2/2^2 + 3/2^3....N/2^N.*/

    Scanner lector = new Scanner(System.in);


    int n;
    float resultado = 0;

    System.out.println("Ingrese numero: ");
     n = lector.nextInt();

      for(int i=1; i<=n; i++){
         
         resultado = resultado+(i/2^i);

          }

           System.out.println("la suma es: " + resultado);
   
  }
}
